src/mc_tasks/lipm_stabilizer/StabilizerTask.cpp

```diff
@@ -574,17 +604,98 @@ void StabilizerTask::staticTarget(const Eigen::Vector3d & com, double zmpHeight)
 void StabilizerTask::target(const Eigen::Vector3d & com,
                             const Eigen::Vector3d & comd,
                             const Eigen::Vector3d & comdd,
-                            const Eigen::Vector3d & zmp)
+                            const Eigen::Vector3d & zmp,
+                            const Eigen::Vector3d & zmpd)
 {
-  comTarget_ = com;
+  comTargetRaw_ = com;
+  comTarget_ = comTargetRaw_ - comOffsetErrCoM_;
+  if(c_.extWrench.addExpectedCoMOffset)
+  {
+    comTarget_ -= comOffsetTarget_;
+  }
   comdTarget_ = comd;
   comddTarget_ = comdd;
   zmpTarget_ = zmp;
+  zmpdTarget_ = zmpd;
   double comHeight = comTarget_.z() - zmpTarget_.z();
   omega_ = std::sqrt(constants::gravity.z() / comHeight);
   dcmTarget_ = comTarget_ + comdTarget_ / omega_;
 }
 
+void StabilizerTask::setExtWrenches(const std::vector<std::pair<std::string, sva::ForceVecd>> & extWrenches)
+{
+  extWrenchesTarget_ = extWrenches;
+  comOffsetTarget_ = computeCoMOffset(extWrenchesTarget_, robot());
+  comTarget_ = comTargetRaw_ - comOffsetErrCoM_;
+  if(c_.extWrench.addExpectedCoMOffset)
+  {
+    comTarget_ -= comOffsetTarget_;
+  }
+}
+
+Eigen::Vector3d StabilizerTask::computeCoMOffset(const std::vector<std::pair<std::string, sva::ForceVecd>> & extWrenches,
+                                                 const mc_rbdyn::Robot & robot) const
+{
+  Eigen::Vector3d comOffset = Eigen::Vector3d::Zero();
+
+  for(const auto & extWrench : extWrenches)
+  {
+    Eigen::Vector3d pos, force, moment;
```

Comment:
Can be moved out of the loop

Reply:
Fixed in https://github.com/jrl-umi3218/mc_rtc/pull/95/commits/acee8bfc329557e61ad0f1ea6f031848a98cab64